sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/RedisPatchSchedulesImpl.java

```diff
@@ -25,15 +26,17 @@
         RedisCache> {
     // Currently Redis Cache has one PatchSchedule
     private final String patchScheduleName = "default";
+    private boolean load = false;
 
     RedisPatchSchedulesImpl(RedisCacheImpl parent) {
         super(parent, parent.taskGroup(), "PatchSchedule");
-        if (parent.id() != null) {
-            this.cacheCollection();
-        }
     }
 
     Map<String, RedisPatchSchedule> patchSchedulesAsMap() {
+        if (!load) {
+            load = true;
+            cacheCollection();
+        }
         Map<String, RedisPatchSchedule> result = new HashMap<>();
```

Comment:
Seems better to put to a shared method like `ensureCollectionLoaded` etc.

Reply:
done